### PR TITLE
Fix queue unit tests 

### PR DESCRIPTION
**Pull request by champly**
#23381

### CODE REVIEW

**Comment by champly** [comment]
@hzxuzhonghu [https://github.com/istio/istio/pull/23286#discussion_r415285740](url) 
how about replacing `time.Sleep` with `time.AfterFunc`?

File: pkg/queue/instance_test.go [hunk 1]
```diff
@@ -101,13 +101,13 @@ func TestResourceFree(t *testing.T) {
...
-
-	close(stop)
+	// mock queue block wait cond signal
+	time.AfterFunc(10*time.Millisecond, func() {
+		close(stop)
+	})
 
 	select {
-	case <-time.After(500 * time.Microsecond):
+	case <-time.After(20 * time.Millisecond):
```
Comment (howardjohn) [inline pkg/queue/instance_test.go:110]: can you make the timeout higher? with 20ms `12627 runs so far, 3 failures` with 200ms `30932 runs so far, 0 failures` using https://godoc.org/golang.org/x/tools/cmd/stress

Comment (champly) [reply pkg/queue/instance_test.go]: thanks, in my local get `39438 runs so far, 0 failures`

**Pull request merged**